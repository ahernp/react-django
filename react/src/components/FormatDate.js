import React from 'react';
import PropTypes from 'prop-types'

export default class FormatDate extends React.Component {
    render() {
        const { dateString } = this.props;
        if (dateString) {
            const options = {weekday: 'short', year: 'numeric', month: 'short', day: '2-digit',
                hour12: false, hour: '2-digit', minute: '2-digit', second: '2-digit'};
            const date = new Date(dateString);
            return (<span>{date.toLocaleString('en-GB', options)}</span>);
        }
        else
            return '';
    }
}

FormatDate.propTypes = {
  dateString: PropTypes.string.isRequired
};
