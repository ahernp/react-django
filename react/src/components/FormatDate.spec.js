import FormatDate from "./FormatDate";

it("renders correctly", () => {
  const wrapper = shallow(
    <FormatDate dateString="2001-01-01" />
  );

  expect(wrapper).toMatchSnapshot();
});
