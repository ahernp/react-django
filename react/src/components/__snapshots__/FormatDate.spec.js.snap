// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<span>
  Mon, Jan 01, 2001, 00:00:00
</span>
`;
